### PR TITLE
test: wait for gs fake container to be ready

### PR DESCRIPTION
**Pull request by nopcoder**
Fix https://github.com/treeverse/lakeFS/issues/6811

### CODE REVIEW

File: pkg/block/gs/main_test.go [hunk 1]
```diff
@@ -69,12 +69,26 @@ func TestMain(m *testing.M) {
...
 	// Create the test client and bucket
-	blockURL := fmt.Sprintf("http://%s/storage/v1/", endpoint)
+	blockURL := fmt.Sprintf("http://%s/storage/v1/", url.PathEscape(endpoint))
 	client, err = storage.NewClient(ctx, option.WithEndpoint(blockURL), option.WithoutAuthentication())
 	if err != nil {
 		log.Fatalf("Could not create gs client: %s", err)
 	}
 
+	// Wait for the container to be ready by listing buckets
+	if err := pool.Retry(func() error {
```
Comment (N-o-Z) [inline pkg/block/gs/main_test.go:79]: Can we please explicitly define the max wait?

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Looks good, one comment

File: pkg/block/gs/main_test.go (hunk 1 above)
Comment (nopcoder) [reply pkg/block/gs/main_test.go:79]: The default is 1 minute - which is more than enough for this issue.

Comment (N-o-Z) [reply pkg/block/gs/main_test.go:79]: I'm sure it's more than enough - I think that for visibility sake, it makes sense to explicitly define it in dockerpool setup

Comment (nopcoder) [reply pkg/block/gs/main_test.go:79]: Less code == less bugs

Comment (N-o-Z) [reply pkg/block/gs/main_test.go:79]: Very well - not a hill I'm will to die on

**Review by N-o-Z — APPROVED** [review]

**Pull request merged**